### ISSUE TITLE
[Accessibility] Empty aria-label for Search Icon on search input "Search post by keyword"

### ISSUE THREAD

**Issue by kivuvarosekivuvan** [ushahidi/platform#4977]
## Describe the bug
- The search icon <mat-icon>, which is inside the <mzima-client-button> component, has an empty or missing aria-label attribute. This makes it challenging for users, especially those using screen readers, to understand the purpose of the icon.

### Where to find the code:
- https://github.com/ushahidi/platform-client-mzima/blob/development/apps/web-mzima-client/src/app/shared/components/search-form/search-form.component.html

## Where was the bug observed
- The bug was observed on the /map page at http://localhost:4200/map, specifically on the search input field which also contains the search icon.

## To Reproduce
1. Navigate to http://localhost:4200/map.
2. Inspect the search icon using a screen reader or your browser's developer tools.
3. Observe that the search icon does not have an aria-label attribute.

## Expected behavior
- The search icon should have a meaningful aria-label attribute that clearly describes the purpose of the icon, making it accessible for all users 

## Walk through

https://github.com/user-attachments/assets/2c0d1777-2095-4ffc-ba59-f15ad466112d

<img width="1178" alt="Screenshot 2024-07-15 at 12 59 20" src="https://github.com/user-attachments/assets/21a59395-2ab3-4974-a9e4-498f75f0250b">
<img width="1595" alt="Screenshot 2024-07-15 at 13 01 03" src="https://github.com/user-attachments/assets/b7fe4efb-11e9-47fd-bc2d-139955ea1e26">
